subprojects/instant-execution-report/src/main/kotlin/data/ListExtensions.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package data
+
+
+fun <T> List<T>.updateElementAt(index: Int, update: (T) -> T): List<T> = mutableListOf<T>().also { result ->
```

Comment:
:o: I find this name confusing as the function doesn't "update the element at" but "creates a copy with the element at `index` updated"